doc/developer-guides/hld/mmio-dev-passthrough.rst

```diff
@@ -0,0 +1,34 @@
+.. _mmio-device-passthrough:
+
+MMIO Device Passthrough
+##################
+
+The ACRN Hypervisor not only supports PCI device passthrough, but also supports
+MMIO device passthrough. However, there are some constraints about the passthrough
+MMIO device. The ACRN hypervisor assumes there're no DMA access to the MMIO device
+and the MMIO device doesn't use IRQ.
+
+The ACRN supports the MMIO device passthrough in the following way:
+a) For pre-launched VM, the ACRN hypervisor use the VM configuration to tell
+where're the physical MMIO regions and where we want to map the MMIO regions
+to in pre-launched VM. Then the ACRN hypervisor will map these MMIO regions to
+the pre-launched VM and remove these MMIO regions from SOS. Besides this,
+the ACRN hypervisor needs to fill the vACPI table for this MMIO device
+bases on the physical ACPI Table of this device.
+2)For post-launched VM, besides what've done for pre-launched VM, we alse need
+to use the cmdline to tell which MMIO device we should pass through to a
+post-launched VM. Now we have two ways: a) If the MMIO device only has MMIO regions
+and no ACPI Table touched, using "--mmiodev_pt MMIO_regions", b) If the MMIO device
+touches ACPI Table, using "--acpidev_pt HID". Now we only support TPM2 MSFT0101
+pass through by "--acpidev_pt MSFT0101". However, the HID for the passthrough TPM
+and vTPM are the same. For now, if we want to pass through a TPM to a post-launched
+VM, we will ignore the vTPM emulation request.
```

Comment:
We should not ignore vTPM. It is up to user configuration

Reply:
Now the vTPM and the PT TPM have the same HID and the same MMIO regions. The PT needs to access the MMIO regions
directly. However, the vTPM accesses the MMIO regions should trap and emulate. How should we handle this ?